internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestStoreMigration.java

```diff
@@ -325,6 +325,7 @@ public void testStoreMigrationWithDaVinciPushStatusSystemStore() throws Exceptio
               .assertEquals(pushStatusStoreReader.getPartitionStatus(storeName, 1, 0, Optional.empty()).size(), 1));
 
       startMigration(parentControllerUrl, storeName);
+      checkMigrationStatus(parentControllerUrl, storeName);
```

Comment:
Referring to comment about main():

We don't have to sweat main or global variables now that we have a print function closure.  We can change the signature of checkMigrationStatus which takes a PrintFunction we define here.  Our PrintFunction in our test would look like:

      List<String> statusOutput = new ArrayList<>();
      AdminTool.PrintFunction printFunction = (string) -> {statusOutput.add(string)};
      checkMigrationStatus(parentControllerUrl, storeName,printFunction);
      Assert.assertTrue(statusOutput.contains(storeName));
      // assert ssystem storeName etc.

No global variable needed, no execution of main needed.  Admittedly a full integration test would go end to end with main, but I think this is ok for now.

Reply:
Thank you for the clarification regarding the closure usage.

For this are we bypassing main? Since `printFunction` isn't a string I can't pass it into the function below as an argument inside the test file.

```
private void checkMigrationStatus(String controllerUrl, String storeName) throws Exception {
    String[] checkMigrationStatusArgs = { "--migration-status", "--url", controllerUrl, "--store", storeName,
        "--cluster-src", srcClusterName, "--cluster-dest", destClusterName };
    AdminTool.main(checkMigrationStatusArgs);
}
```

If so, would it be ok to make `checkMigrationStatus` in AdminTool.java a public function so I can directly invoke it in the test file?